File: problem.js
function maxSubarrayProduct(nums, k, m) {
    const n = nums.length;
    const dpmax = Array.from({ length: n + 1 }, () => Array(m + 1).fill(0));
    const dpmin = Array.from({ length: n + 1 }, () => Array(m + 1).fill(0));
    let result = Number.MIN_SAFE_INTEGER;

    for (let i = 1; i <= n; ++i) {
        dpmax[i][1] = nums[i - 1];
        dpmin[i][1] = nums[i - 1];
        for (let j = 2; j <= Math.min(m, i); ++j) {
            let currentMax = nums[i - 1];
            let currentMin = nums[i - 1];
            for (let l = i - 1; l >= Math.max(1, i - k + 1); --l) {
                currentMax = Math.max(currentMax, Math.max(nums[l - 1], dpmax[l][j - 1] * nums[i - 1]));
                currentMin = Math.min(currentMin, Math.min(nums[l - 1], dpmin[l][j - 1] * nums[i - 1]));
            }
            dpmax[i][j] = currentMax;
            dpmin[i][j] = currentMin;
        }
    }

    for (let i = k; i <= n; ++i) {
        for (let j = k; j <= m; ++j) {
            result = Math.max(result, dpmax[i][j]);
        }
    }

    return result;

File: problem.py
def max_subarray_product(nums, k, m):
    n = len(nums)
    dpmax = [[0] * (m + 1) for _ in range(n + 1)]
    dpmin = [[0] * (m + 1) for _ in range(n + 1)]
    result = float('-inf')

    for i in range(1, n + 1):
        dpmax[i][1] = nums[i - 1]
        dpmin[i][1] = nums[i - 1]
        for j in range(2, min(m, i) + 1):
            current_max, current_min = nums[i - 1], nums[i - 1]
            for l in range(i - 1, max(0, i - k), -1):
                current_max = max(current_max, max(nums[l - 1], dpmax[l][j - 1] * nums[i - 1]))
                current_min = min(current_min, min(nums[l - 1], dpmin[l][j - 1] * nums[i - 1]))
            dpmax[i][j] = current_max
            dpmin[i][j] = current_min

    for i in range(k, n + 1):
        for j in range(k, m + 1):
            result = max(result, dpmax[i][j])

    return result

nums = [2, -5, -2, -4, 3]
k, m = 2, 3
print(max_subarray_product(nums, k, m))
